tests/drivers/i2c/i2c_emul/src/test_forwarding_pio.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "emulated_target.hpp"
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+
+namespace
```

Comment:
why an anonymous namespace here?

Reply:
Anonymous namespaces have a few advantages over declaring everything static. https://medium.com/pranayaggarwal25/unnamed-namespaces-static-f1498741c527 is a pretty good read. The closing comment is similar to the closing comment we put on long #ifdef so if we have nested namespaces it's easier to follow which one is closed.

Reply:
I feel like I'm going to have to learn more and more C++ to keep reviewing this, and I'd say this is a problem as I don't have any inclination, time, or energy to learn C++ in depth any time soon. This is problematic for me as the current maintainer of i2c. I think if closures are the key feature you are looking for from C++ sticking with that, and ensuring some solid comments around them is pretty crucial. Beyond that keeping C++ usage to a minimum would be my recommendation.

If I'm getting the gist of this though its an alternative to labeling variables/functions static to keep them private to the compilation unit. Except... the entire rest of the Zephyr code base uses this already. So why are we doing things different for one set of unit tests?

Reply:
Seems ok I suppose, not -1 worthy but I don't necessarily love this either.